src/scope/core/pipelines/schema.py

```diff
@@ -7,18 +7,53 @@
 - API introspection and automatic UI generation
 
 Pipeline-specific configs inherit from BasePipelineConfig and override defaults.
+Each pipeline defines its supported modes and can provide mode-specific defaults.
 """
 
-from typing import Annotated, Any, ClassVar
+from typing import Annotated, Any, ClassVar, Literal
 
 from pydantic import BaseModel, ConfigDict, Field
 
+# Type alias for input modes
+InputMode = Literal["text", "video"]
+
+
+class ModeDefaults(BaseModel):
+    """Mode-specific default values.
+
+    These override the base config defaults when operating in a specific mode.
+    Only non-None values will override the base defaults.
+    """
+
+    model_config = ConfigDict(extra="forbid")
+
+    # Resolution can differ per mode
+    height: int | None = None
+    width: int | None = None
+
+    # Core parameters
+    denoising_steps: list[int] | None = None
+
+    # Video mode parameters
+    noise_scale: float | None = None
+    noise_controller: bool | None = None
+    input_size: int | None = None
+
+    # Default prompts for this mode (list of prompt text strings, will be converted to PromptItems with weight 1.0)
+    default_prompts: list[str] | None = None
```

Comment:
This doesn't seem like the right place to define default prompts because for V2V we're defining the default prompt based on a specific default input video that is used in the frontend. But, only the frontend knows what the default input video is. So, it seems strange to define the default prompt here since its making an assumption about what the frontend does. IMO would be better not to make assumptions about how is using the pipeline - the frontend just happens to be the primary consumer right now.